### ISSUE TITLE
ERROR: Digistuff is not allowed to use the HTTP API. Digilines will not be available.

### ISSUE THREAD

**Issue by CHJ85** [mt-mods/digistuff#30]
Hi there. What does this error mean and how do I fix it?
It also says that if this functionality is desired, please add digistuff to your secure.http_mods settings.
How do I do that and why is this a thing?

**Comment by S-S-X** [comment]
Minetest server configuration file, mod names making HTTP requests / sending data to internet must be in `secure.http_mods` configuration. Like in minetest.conf
```
secure.http_mods = digistuff
```

This is Minetest server restriction so that users pay attention, mods added to that configuration list are able to use any internet service sending and receiving data from internet.

**Comment by S-S-X** [comment]
Oh, also allowing unrestricted internet use for mod will enable digistuff:nic which is in game modem.
You can then for example write code with lua controller and digiline screens to read actual real world internet web pages in game.

**Comment by CHJ85** [comment]
Ah I see. Thank you. So it's a security procedure to prevent malware from sneaking in through mods and whatnot? That makes sense.
But I wish that the error message could just tell me to add "secure.http_mods = digistuff" to minetest.conf.

**Comment by SwissalpS** [comment]
> But I wish that the error message could just tell me to add "secure.http_mods = digistuff" to minetest.conf

that's what this means:
> please add digistuff to your secure.http_mods settings.

I think it avoids saying 'minetest.conf' specifically because you can also change that setting from GUI in advanced settings search for 'http_mods' and add the mods there with spaces

**Comment by S-S-X** [comment]
> But I wish that the error message could just tell me to add "secure.http_mods = digistuff" to minetest.conf.

It might actually be better to not have exact action what to do because this has potential problems with security of players.
This is documented somewhere in minetest docs and there's clear warnings about allowing network for mods if you do not trust said mod.

Basically not having too clear message adds requirement: user must find out what it exactly means and by doing that they will also find information about risks.

Well, probably warning could also be added directly to debug log message...
Maybe Minetest should handle these special privileges by itself, possibly having some "mod permissions" dialog when starting game or in ContentDB browser or something.

**Comment by CHJ85** [comment]
Right. To have an in-game solution with proper documentation within the game itself I think would be a much better idea, because people are impatient unfortunately. So if something doesn't work at first, they'll just ditch it. Or they'll might think it's a bug and blame Minetest, which is obviously unfair.
